src/utils.rs

```diff
@@ -93,39 +71,23 @@ pub fn is_valid_ed25519_pubkey(public_key: &Bound<'_, PyAny>) -> PyResult<bool>
 ///
 ///     Returns:
 ///         True if the address is a valid destination address, False otherwise.
-#[pyfunction]
-pub fn is_valid_bittensor_address_or_public_key(address: &Bound<'_, PyAny>) -> PyResult<bool> {
-    Python::with_gil(|_py| {
-        if address.is_instance_of::<PyString>() {
-            let address_str = &address.to_string();
-            if address_str.starts_with("0x") {
-                is_valid_ed25519_pubkey(address)
-            } else {
-                is_valid_ss58_address(address_str)
-            }
-        } else if address.is_instance_of::<PyBytes>() {
-            is_valid_ed25519_pubkey(address)
+pub fn is_valid_bittensor_address_or_public_key(address: &str) -> bool {
+    if address.starts_with("0x") {
+        // Convert hex string to bytes
+        if let Ok(bytes) = hex::decode(&address[2..]) {
+            is_valid_ed25519_pubkey(&bytes)
         } else {
-            Ok(false)
+            false
         }
-    })
+    } else {
+        is_valid_ss58_address(address)
+    }
 }
 
 pub fn print(s: String) {
-    Python::with_gil(|py| {
-        let locals = PyDict::new_bound(py);
-        locals.set_item("s", s).unwrap();
-        py.run_bound(
-            r#"
-import sys
-print(s, end='')
-sys.stdout.flush()
-"#,
-            None,
-            Some(&locals),
-        )
-        .unwrap();
-    });
+    use std::io::{self, Write};
+    print!("{}", s);
+    io::stdout().flush().unwrap();
```

Comment:
Do we even need this function at all? Why don't we just use `print!` directly?

Reply:
It's part of the existing code that needs to be refactored . First step would be getting the binary to compile so wombo can consume it in their rust wallet